panther_utils/panther_utils/arguments.py

```diff
@@ -0,0 +1,175 @@
+        configuration_data = yaml_data.get("configuration", {})
+        wheel_type = configuration_data.get("wheel_type", "")
+
+        if robot_model not in VALID_CONFIGURATIONS:
+            raise ValueError(
+                f"Invalid robot model '{robot_model}'. "
+                f"Valid models are: {', '.join(VALID_CONFIGURATIONS.keys())}"
+            )
+
+        valid_wheel_types = VALID_CONFIGURATIONS[robot_model]["wheel_type"]
+        if wheel_type not in valid_wheel_types:
+            raise ValueError(
+                f"Invalid wheel type '{wheel_type}' for {robot_model}. "
+                f"Valid wheel types are: {', '.join(valid_wheel_types)}"
+            )
+
+    def normalize_robot_configuration(self, yaml_data: dict) -> dict:
+        """Normalizes the YAML dictionary structure to a flat to nested format and validates the structure."""
+
+        obligatory_keys = {"configuration"}
+
+        # Checking and normalize flat structure
+        if obligatory_keys.issubset(yaml_data.keys()):
+            self.validate_robot_configuration(yaml_data)
+            return {"": yaml_data}
+
+        # Checking nested structure
+        if all(isinstance(value, dict) for value in yaml_data.values()):
+            nested_data = {}
+            for namespace, robot_config in yaml_data.items():
+                if not obligatory_keys.issubset(robot_config.keys()):
+                    raise ValueError(
+                        f"Invalid nested YAML structure at '{namespace}': Missing required keys."
+                    )
+                self.validate_robot_configuration(robot_config)
+                nested_data[namespace] = robot_config
+            return nested_data
+
+        raise ValueError("Invalid YAML structure: The data does not match expected formats.")
+
+    def extract_first_robot_configuration(self, yaml_data: dict) -> Tuple[str, dict]:
+        """Extracts the namespace and configuration of first element in dict."""
+        keys = list(yaml_data.keys())
+        namespace = keys[0]
+        configuration = yaml_data[namespace]
+        return namespace, configuration
+
+    def create_launch_arguments(
+        self, namespace: str, robot_config: dict
+    ) -> list[DeclareLaunchArgument]:
+        """Declare launch arguments based on the YAML configuration files."""
+        robot_model = robot_config.get("robot_model", "panther")
+        x, y, z = robot_config.get("init_pose", [0.0, 0.0, 0.0])
+        roll, pitch, yaw = robot_config.get("init_rotation", [0.0, 0.0, 0.0])
+        x, y, z, roll, pitch, yaw = map(str, [x, y, z, roll, pitch, yaw])
+        configuration_data = robot_config.get("configuration", {})
+        wheel_type = configuration_data.get("wheel_type", "")
+
+        # Prefer declaration over configuration:
+        namespace = LaunchConfiguration("namespace", default=namespace)
+        robot_model = LaunchConfiguration("robot_model", default=robot_model)
```

Comment:
Yes it was, because `DeclerateLaunchArgument` with **default_value** `LaunchConfiguration` takes **default** yaml data only if `ros2cli` (`wheel_type:=...`) argumen is not specify. Anyway `ros2cli` arg is set as argument.

I deleted this function because there was too much correlation.